### ISSUE TITLE
environment: action space + per-vertex legality mask

### ISSUE THREAD

**Issue by BjarkeKarlsen** [BjarkeKarlsen/slap-mapd-coupling#13]
Implement `environment/spaces.py`: the action space and the per-vertex
legality mask (sec:method:rl, eq:mask).

**Spec:** with `d_max = max_{v in V_mov} |N+(v)|`, the policy head has
`d_max + 1` logits (one wait, `d_max` move slots). At a state with
`l_i(t)=v`, logits for slots beyond `|N+(v)|` (i.e. for
`U_i \ U(v)`) are set to `-infinity` before the softmax:

```
Pr[u_i(t) = .] = softmax(l(.) + M(v,.))
M(v,.) = 0        if . in U(v)
M(v,.) = -infty   if . in U_i \ U(v)
```

This is what makes each individual action legal *by construction*, not in
expectation — distinct from joint-action legality, which is the conflict-
resolution operator's job (#conflict-resolution operator).

Note (thesis, explicit caveat): RLlib's newer multi-agent API does not ship
action masking as a built-in; it needs a custom module or connector. Even
though this issue is filed under M2 (the mask is needed by every controller,
not just the decentralised one), budget the RLlib-specific plumbing as part
of #action masking (RLlib custom module/connector) under M5, not here — this
issue is the *mask itself* (shape, legality logic), independent of RLlib.
